TicTacToe/Program.cs

```diff
@@ -7,415 +7,96 @@
-                //pos[7] = " ";
-                //pos[8] = " ";
-                //pos[9] = " ";
-                return true;
-            }
-            if (WinLine(1, 4, 7, "O")) //O wins vertically
-            {
-                //pos[1] = " ";
-                pos[2] = " ";
-                pos[3] = " ";
-                //pos[4] = " ";
-                pos[5] = " ";
-                pos[6] = " ";
-                //pos[7] = " ";
-                pos[8] = " ";
-                pos[9] = " ";
-                return true;
-            }
-            if (WinLine(2, 5, 8, "O"))
-            {
-                pos[1] = " ";
-                //pos[2] = " ";
-                pos[3] = " ";
-                pos[4] = " ";
-                //pos[5] = " ";
-                pos[6] = " ";
-                pos[7] = " ";
-                //pos[8] = " ";
-                pos[9] = " ";
-                return true;
-            }
-            if (WinLine(3, 6, 9, "O"))
-            {
-                pos[1] = " ";
-                pos[2] = " ";
-                //pos[3] = " ";
-                pos[4] = " ";
-                pos[5] = " ";
-                //pos[6] = " ";
-                pos[7] = " ";
-                pos[8] = " ";
-                //pos[9] = " ";
-                return true;
-            }
-            if (WinLine(1, 5, 9, "O")) //O wins diagonally
-            {
-                //pos[1] = " ";
-                pos[2] = " ";
-                pos[3] = " ";
-                pos[4] = " ";
-                //pos[5] = " ";
-                pos[6] = " ";
-                pos[7] = " ";
-                pos[8] = " ";
-                //pos[9] = " ";
-                return true;
-            }
-            if (WinLine(3, 5, 7, "O"))
-            {
-                pos[1] = " ";
-                pos[2] = " ";
-                //pos[3] = " ";
-                pos[4] = " ";
-                //pos[5] = " ";
-                pos[6] = " ";
-                //pos[7] = " ";
-                pos[8] = " ";
-                pos[9] = " ";
-                return true;
-            }
-            else
-            {
-                return false;
-            }
-        }
-
+    {
          static void Main()
          {
-            StartGame: //restart from end of Main
-            
-            /* variable declaration:
-             * choice: selects where on board
-             * turn: toggles between x and o
-             * num: counts number of turns
-            */
-
-            int choice = 0, turn = 1, num = 1;
-            bool winFlag = false, playing = true;
-
-            DrawBoard(); //show players board
-
-            //game loop
-            while (playing == true)
+            while (true)
             {
-               
+                var winningPlayer = -1; //-1 null, 0 tie, 1 player 1 (X), 2 player 2 (O)
+                var totalTurns = 0;
+                var isGameOver = false;
+                var board = new Board();
```

Comment:
Rather than have program.cs manage all the complexity of a digital representation of this game, abstract out the complexity further into Objects that you model.

Program depends on Board which depends on BoardPosition. BoardPosition is just a manager of the actual state of the board. Board manages everything like else like how to print the state of the board and check if the board is in a winning state.

They could honestly remain as one object, but I felt like breaking it out.